code/modules/mob/mob_emote.dm

```diff
@@ -148,9 +148,10 @@
 				if(isturf(oldloc) && isturf(newloc))
 					user.SpinAnimation(5, 1)
 					user.glide_for(0.6 SECONDS) // This and the glide_for below are purely arbitrary. Pick something that looks aesthetically pleasing.
-					user.forceMove(newloc)
+					user.pass_flags += PASSMOB | PASSTABLE
+					step(user,get_dir(oldloc,newloc))
+					user.pass_flags -= PASSMOB | PASSTABLE
```

Comment:
what if they have passmob flag already? or pass table?
cashe their flags before adding the new ones. then reapply the cash afterwards.
also use the `|=` operator to add bitflags. 
if you had a bitflag value of 2 then you added the 2 bitflag again then it wouldnt have the 2 bitflag anymore it would have the 4 bitflag.